### PR TITLE
Hide empty zones

### PR DESCRIPTION
**Pull request by ivannaranjo**
This PR hides the empty zones from the GCE listings, including those zones that become empty when the user filters out the non-Windows machines.

The PR accomplishes this but doing a bit of refactoring on how the zone view models are created, and how the queries that process the data to refresh the UI.

There's also a small cosmetic change, in which the GCE instances, or the zones, are sorted alphabetically.

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension/CloudExplorerSources/Gce/GceSourceRootViewModel.cs [hunk 1]
```diff
@@ -176,6 +176,11 @@ private void OnNewInstanceCommand()
             Process.Start(url);
         }
 
+        private void OnStatusCommand()
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/CloudExplorerSources/Gce/GceSourceRootViewModel.cs]: No need to do this now, but we should make this a util somewhere.  I know we do this in many places.

File: GoogleCloudExtension/GoogleCloudExtension/CloudExplorerSources/Gce/GceSourceRootViewModel.cs [hunk 2]
```diff
@@ -226,73 +231,43 @@ private GceDataSource CreateDataSource()
 
         private void PresentViewModels()
         {
+            IEnumerable<TreeNode> viewModels;
             if (_showZones)
             {
-                PresentZoneViewModels();
+                viewModels = _instancesPerZone
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/CloudExplorerSources/Gce/GceSourceRootViewModel.cs]: This is a little hard to follow as it is such a large query, I would consider breaking it up into parts but up to you

**Comment by iantalarico** [comment]
LGTM, nice refactor

File: GoogleCloudExtension/GoogleCloudExtension/CloudExplorerSources/Gce/GceSourceRootViewModel.cs (hunk 2 above)
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/CloudExplorerSources/Gce/GceSourceRootViewModel.cs]: I'll add a comment to it. I tried to make it a step by step query but clearly failed :). Will see what I can do.
